File: Visual Studio/C#/Eruru.MVVM/Shared/Binding/MVVMBindingRelativeSource.cs
using System;

namespace Eruru.MVVM {

	public class MVVMBindingRelativeSource {

		public MVVMRelativeSourceMode Mode { get; }
		public Type AncestorType { get; }
		public int AncestorLevel { get; }

		public MVVMBindingRelativeSource () {
			Mode = MVVMRelativeSourceMode.Self;
		}
		public MVVMBindingRelativeSource (Type ancestorType) : this (ancestorType, 1) {

		}
		public MVVMBindingRelativeSource (int ancestorLevel) : this (null, ancestorLevel) {

		}
		public MVVMBindingRelativeSource (Type ancestorType, int ancestorLevel) {
			AncestorType = ancestorType;
			AncestorLevel = ancestorLevel;
			Mode = MVVMRelativeSourceMode.FindAncestor;
		}

	}

}

File: WebStorm/JavaScript/eruru.mvvm/MVVMBindingRelativeSource.js
function MVVMBindingRelativeSource () {

	this.mode = null;
	this.ancestorType = null;
	this.ancestorLevel = 1;

	this.autoSetMode = function () {
		if (this.ancestorType != null) {
			this.mode = MVVMBindingRelativeSourceMode.findAncestor;
			return;
		}
		this.mode = MVVMBindingRelativeSourceMode.self;
	};

}